Sources/BrightFutures/Result+BrightFutures.swift

```diff
@@ -8,6 +8,17 @@
 
 import Result
 
+extension ResultProtocol {
```

Comment:
Is there a reason that we're adding this here? From what I can tell it doesn't look like this needs to be here in order to fix the issue at hand but that's just from a cursory glance at the Result code. If no, it might be better to put this up on the Result repo rather than this one because I do like the addition

Reply:
Sorry, I need to learn how to read PR descriptions 😅. I do still think that you should put up a PR for this on Result's repo but this is a nice stop gap measure.

Reply:
As you can see, Result framework now encapsulate a Result in a result(resultProtocol)
My others change demonstrate it.

Result framework do not want all this function on the protocol, that’s not the spirit of their modications (composition pattern)

I make it as extension here just to not edit a lot of code, ie. do xxx.result.analysis everywhere